src/employees.service.ts

```diff
@@ -0,0 +1,147 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, Observable, of, throwError } from 'rxjs';
+import { delay } from 'rxjs/operators';
+import * as employeeHier from './employees.json';
+
+const CREATE_ACTION = 'create';
+const UPDATE_ACTION = 'update';
+
+const makeFlatData = (data, idField, parentField, containsField) => {
+    const result = [];
+    function visitChildren(childrenArray: any[], parentId: any): void {
+        if (!childrenArray ) {
+            return;
+        }
+        for (let obj of childrenArray) {
+            let blank = {};
+            blank[parentField] = parentId;
+            result.push(Object.assign(blank, obj));
+            visitChildren(obj[containsField], obj[idField]);
+        }
+    }
+    visitChildren([data[0]], null);
+    return result;
+};
+
+export interface Employee {
+    id: number;
+    managerId: number;
+    name: string;
+    firstName: string;
+    reports: any;
+    lastName: string;
+    title: string;
+    active: boolean;
+    phone: number;
+    hireDate: Date;
+}
+export function createEmployee(id: number, managerId: number, firstName: string, lastName: string, title: string, phone: number, active: boolean = true, reports: any = [],  hireDate: Date = null): Employee {
+    const emp: Employee = {
+        id: id,
+        managerId: managerId,
+        name: `${firstName} ${lastName}`,
+        firstName: firstName,
+        reports: reports,
+        lastName: lastName,
+        title: title,
+        active: active,
+        phone: phone,
+        hireDate: new Date(hireDate)
+    };
+    return emp;
+}
+
+export const employeeTitles = ["Chief Executive Officer",
+        "Chief Technical Officer",
+        "VP, Engineering",
+        "Team Lead",
+        "Software Architect",
+        "Director, Engineering",
+        "QA Architect", "Chief Product Officer"];
+
+export function makeEmployees(data: any[] , containsField: string
```

Comment:
This doesn't actually make anything—it parses dates. And it does it in place, by mutating data.
It's better to copy it instead so different instances of the service always get identical source data.